File: gui_view/src/main/scala/com/goexp/galgame/gui/view/game/detailview/part/CVSearchController.java
package com.goexp.galgame.gui.view.game.detailview.part;

import com.goexp.galgame.common.website.BangumiURL;
import com.goexp.galgame.common.website.WikiURL;
import com.goexp.galgame.gui.util.Websites;
import com.goexp.galgame.gui.view.DefaultController;
import javafx.fxml.FXML;
import javafx.scene.control.MenuItem;


public class CVSearchController extends DefaultController {

    private String keyword = "";

    @FXML
    private MenuItem linkDlsite;

    @FXML
    private MenuItem linkWiki;

    @FXML
    private MenuItem link2DF;

    @FXML
    private MenuItem linkBangumi;

    protected void initialize() {

        linkWiki.setOnAction((e) -> Websites.open(WikiURL.fromTitle(keyword)));
        linkBangumi.setOnAction((e) -> Websites.open(BangumiURL.fromTitle(keyword)));
    }

    public void load(String keyword) {
        this.keyword = keyword;
    }


}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/view/game/detailview/part/CVSearchController.scala
package com.goexp.galgame.gui.view.game.detailview.part

import com.goexp.galgame.common.website.{BangumiURL, WikiURL}
import com.goexp.galgame.gui.util.Websites
import com.goexp.ui.javafx.DefaultController
import javafx.fxml.FXML
import javafx.scene.control.MenuItem

class CVSearchController extends DefaultController {
  private var keyword = ""
  @FXML private var linkWiki: MenuItem = _
  @FXML private var linkBangumi: MenuItem = _

  override protected def initialize() = {
    linkWiki.setOnAction(_ => Websites.open(WikiURL.fromTitle(keyword)))
    linkBangumi.setOnAction(_ => Websites.open(BangumiURL.fromTitle(keyword)))
  }

  def load(keyword: String): Unit = this.keyword = keyword
}